hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/slowlog/SlowLogEventHandler.java

```diff
@@ -0,0 +1,137 @@
+ */
+@InterfaceAudience.Private
+class SlowLogEventHandler implements EventHandler<RingBufferTruck> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SlowLogEventHandler.class);
+
+  private final SlowLogPayload[] slowLogPayloads;
+
+  private AtomicInteger slowLogPayloadIdx = new AtomicInteger();
+
+  private static final Lock LOCK = new ReentrantLock();
+
+  private static final int WRITE_LOCK_WAIT = 1;
+  private static final int READ_LOCK_WAIT = 1;
+
+  SlowLogEventHandler(int eventCount) {
+    this.slowLogPayloads = new SlowLogPayload[eventCount];
+  }
+
+  /**
+   * Called when a publisher has published an event to the {@link RingBuffer}
+   *
+   * @param event published to the {@link RingBuffer}
+   * @param sequence of the event being processed
+   * @param endOfBatch flag to indicate if this is the last event in a batch from
+   *   the {@link RingBuffer}
+   * @throws Exception if the EventHandler would like the exception handled further up the chain
+   */
+  @Override
+  public void onEvent(RingBufferTruck event, long sequence, boolean endOfBatch) throws Exception {
+    SlowLogPayload slowLogPayload = event.getPayload();
+    if (LOG.isTraceEnabled()) {
+      LOG.trace("Received Slow Log Event. RingBuffer sequence: {}, isEndOfBatch: {}, " +
+          "Event Call: {}", sequence, endOfBatch,
+        slowLogPayload.getCallDetails());
+    }
+    try {
+      if (LOCK.tryLock(WRITE_LOCK_WAIT, TimeUnit.SECONDS)) {
+        int index = slowLogPayloadIdx.getAndSet(
+          (slowLogPayloadIdx.get() + 1) % slowLogPayloads.length);
+        slowLogPayloads[index] = slowLogPayload;
+      }
+    } finally {
+      LOCK.unlock();
+    }
+  }
+
+  /**
+   * Cleans up slow log payloads
+   *
+   * @return true if slow log payloads are cleaned up, false otherwise
+   */
+  boolean clearSlowLogs() {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Received request to clean up online slowlog buffer..");
+    }
+    boolean isCleanedUp = true;
+    try {
+      if (LOCK.tryLock(WRITE_LOCK_WAIT, TimeUnit.SECONDS)) {
+        for (int i = 0; i < slowLogPayloads.length; i++) {
+          slowLogPayloads[i] = null;
+        }
+        slowLogPayloadIdx.set(0);
+      }
+    } catch (InterruptedException e) {
+      isCleanedUp = false;
+      LOG.error("Error while cleaning up slow logs buffer...", e);
+    } finally {
+      LOCK.unlock();
+    }
+    return isCleanedUp;
+  }
+
+  /**
+   * Retrieve list of slow log payloads
+   *
+   * @return list of slow log payloads
+   */
+  List<SlowLogPayload> getSlowLogPayloads() {
+    List<SlowLogPayload> slowLogPayloadList = null;
+    try {
+      if (LOCK.tryLock(READ_LOCK_WAIT, TimeUnit.SECONDS)) {
+        slowLogPayloadList = Arrays.stream(slowLogPayloads)
+          .filter(Objects::nonNull)
+          .collect(Collectors.toList());
+      }
+    } catch (InterruptedException e) {
+      LOG.error("Error while reading slow logs buffer...", e);
```

Comment:
Is swallowing the interrupt the right thing to do? Someone is requesting this thread terminate...